docs/core/profiles.md

```diff
@@ -0,0 +1,394 @@
+`10m` or a maximum vehicle weight of `3.5tons` is zero (these edges are blocked).
+
+#### Customizing `distance_influence`
+
+`distance_influence` allows you to control the trade-off between a fast route (minimum time) and a short route
+(minimum distance). Setting it to `0` means that GraphHopper will return the fastest possible route, *unless* you set 
+the priority of any edges to something different than `1.0`. Using the priority factors will always to some part favor
+some routes because they are shorter as well. Higher values of `distance_influence` will prioritize routes that are fast
+(but maybe not fastest), but at the same time are short in distance.
+
+You use it like this:
+```yaml
+distance_influence: 100
+``` 
+
+The default is `70`. More precisely, by specifying the `distance_influence` you tell the routing engine how much time
+you need to save on a detour (a longer distance route) such that you prefer taking the detour compared to a shorter 
+distance route. Assuming that all priorities are `1` a value of zero means that no matter how little time you can save 
+when doing a detour you will take it, i.e. you always prefer the fastest route. A value of `30` means that one extra
+kilometer must save you `30s` of travelling time. Or to put it another way if a reference route takes `600s` and is 
+`10km` long, `distance_influence=30` means that you are willing to take an alternative route that is `11km` long only if
+it takes no longer than `570s` (saves `30s`).      
+
+## Setting up a Custom Profile
+
+Custom profiles work very much the same way as the 'standard' profiles described at the beginning of this document. The
+custom model needs to be written in a separate YAML file and the `weighting` has to be set to `custom`. Use the 
+`custom_model_file` field to point to the custom model file like this:
+
+```yaml
+profiles:
+  - name: my_custom_profile
+    vehicle: car
+    weighting: custom
+    custom_model_file: path/to/my_custom_profile.yaml
+  
+```
+
+Selecting a custom profile also works like selecting a standard profile. In our example just add `profile=my_custom_profile`
+to your routing request. To set up hybrid- or speed-mode, simply use the `profiles_ch/lm` sections and add the name of
+your custom profile (just like you do for standard profiles).
+
+## Inheritance between Custom Profiles
+
+When specifying a custom profile it is possible to use another custom profile as base 'vehicle', something like:
+```yaml
+profiles:
+  - name: my_custom_profile
+    vehicle: car
+    weighting: custom
+    custom_model_file: path/to/my_custom_profile.yaml
+  - name: my_other_profile
+    vehicle: my_custom_profile
+    weighting: custom
+    custom_model_file: path/to/my_other_custom_profile.yml
+```
+
+In this case the custom model of `my_other_profile` will be determined by merging the two custom models.
+
+todonow: explain merging rules
```

Comment:
@karussell Can you explain the merging rules here?

Reply:
Will do! Really comprehensive docs already, nice!

Reply:
This was a bit tricky to implement. But now it is possible to have a smaller vehicle on the server side and query a bigger one and maintain the optimality of landmark algo.

Please note that this "merging feature" is not yet working for purely server-side custom models. The base is always considered as server-side and the other part is always the custom  model from the query. We could add this feature but I would keep it like this for 1.0 for simplicity.

Reply:
> Please note that this "merging feature" is not yet working for purely server-side custom models.

You mean we cannot merge just *any* models but one always has to be the base? Or what is the difference between 'purely server-side' and using a custom model from the query? Or do you mean we do not allow using one custom profile as base for another at all?

Reply:
Currently it won't work to have two custom models defined in the profile section that depend on each other. But it works to define a custom model in the profile section (server-side) that then can be used in a custom request.

Reply:
Ah ok so the 'inheritance of custom models' is not supported (there is still this header in `profiles.md`)? But what is the difference in terms of merging the custom models by the way?

Reply:
Ah, I thought when using a custom request with a custom model as base, then this is also some kind of inheritance but "on-the-fly"?

> But what is the difference in terms of merging the custom models by the way?

There is no difference, except when we support this on the server-side the user could expect that arbitrary deep inheritance trees like A->B->C-E and B->D would work (and I can imagine importing this properly is not that simple :) ). That was the main reason I left it out for now.

Reply:
Oh ok then. I will adjust the docs.